math/polynomial/polynomial.go

```diff
@@ -0,0 +1,123 @@
+// Package polynomial provides representations of polynomials over the scalars
+// of a group.
+package polynomial
+
+import (
+	"fmt"
+
+	"github.com/cloudflare/circl/group"
+)
+
+// Polynomial stores a polynomial over the set of scalars of a group.
+type Polynomial struct {
+	// Internal representation is in polynomial basis:
+	// Thus,
+	//     p(x) = \sum_i^k c[i] x^i,
+	// where k = len(c)-1 is the degree of the polynomial.
+	c []group.Scalar
+}
+
+// New creates a new polynomial given its coefficients in ascending order.
+// Thus,
+//     p(x) = \sum_i^k c[i] x^i,
+// where k = len(c)-1 is the degree of the polynomial.
+func New(coeffs []group.Scalar) Polynomial {
+	if len(coeffs) == 0 {
+		panic("polynomial: invalid length")
+	}
+	return Polynomial{coeffs}
+}
+
+func (p Polynomial) Degree() uint { return uint(len(p.c) - 1) }
```

Comment:
What if `New` is called with zeroes on the higher coefficients?

Reply:
updated to consider this case